container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+// by stampInfoFile.
+func stamp(inp string) (string, error) {
+	if len(stampInfoFile) == 0 || inp == "" {
+		return inp, nil
+	}
+	formatArgs := make(map[string]interface{})
+	for _, infofile := range stampInfoFile {
+		f, err := os.Open(infofile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to open file %s", infofile)
+		}
+		defer f.Close()
+		// scanner reads line by line and discards '\n' character already
+		scanner := bufio.NewScanner(f)
+		var temp []string
+		for scanner.Scan() {
+			temp = strings.Split(scanner.Text(), " ")
+			key, val := temp[0], temp[1]
+			if _, ok := formatArgs[key]; ok {
+				fmt.Printf("WARNING: Duplicate value for key %s: using %s", key, val)
+			}
+			formatArgs[key] = val
+		}
+		if err = scanner.Err(); err != nil {
+			return "", errors.Wrapf(err, "failed to read line from file %s", infofile)
+		}
+	}
+	// do string manipulation in order to mimic python string format.
+	// specifically, replace '{' with '{{.' and '}' with '}}'.
+	inpReformatted := strings.ReplaceAll(inp, "{", "{{.")
+	inpReformatted = strings.ReplaceAll(inpReformatted, "}", "}}")
+	return formatWithMap(inpReformatted, formattedString(formatArgs)), nil
+}
+
+// mapToKeyValue reverses a map to a '='-separated array of strings in {key}={value} format
+func mapToKeyValue(kvMap map[string]string) []string {
+	keyVals := []string{}
+	concatenated := ""
+	for k, v := range kvMap {
+		concatenated = k + "=" + v
+		keyVals = append(keyVals, concatenated)
+	}
+	return keyVals
+}
+
+// resolveVariables resolves the environment variables embedded in the given value using
+// provided map of environment variable expansions.
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
+		case string:
+			return environment[p]
+		}
+		errorMet = true
+		return ""
+	}
+	if errorMet {
+		err := errors.New("environment variable sought after does not exist in evnrionemnet map")
+		if err != nil {
+			errors.Wrap(err, "failed to create new error")
+		}
+		return "", err
+	}
+	return os.Expand(value, mapper), nil
+}
+
+func main() {
+	flag.Var(&labelsArray, "labels", "Augment the Label of the previous layer.")
+	flag.Var(&ports, "ports", "Augment the ExposedPorts of the previous layer.")
+	flag.Var(&volumes, "volumes", "Augment the Volumes of the previous layer.")
+	flag.Var(&entrypointPrefix, "entrypointPrefix", "Prefix the Entrypoint with the specified arguments.")
+	flag.Var(&env, "env", "Augment the Env of the previous layer.")
+	flag.Var(&command, "command", "Override the Cmd of the previous layer.")
+	flag.Var(&entrypoint, "entrypoint", "Override the Entrypoint of the previous layer.")
+	flag.Var(&layer, "layer", "Layer sha256 hashes that make up this image.")
```

Comment:
Name this variable `layerDigests` to be explicit about what it stores. For the flag, I would suggest `layerDigest`. I would also mention in the help text that the order the layer digests are specified in is important.

Or is this path to files storing the digest of the layer? In that case, name the variable layerDigestFiles, ...

Reply:
it's a single layer digest. This create image config is called in a loop in the bzl, so we don't need to worry about order in this binary as we assume it is handed correctly to us.